src/main/java/com/denizenscript/denizencore/scripts/commands/core/MongoCommand.java

```diff
@@ -0,0 +1,547 @@
+    // @Usage
+    // Change Databases.
+    // - ~mongo id:name use_database:my_new_database
+    //
+    // @Usage
+    // Change Collections.
+    // - ~mongo id:name use_collection:my_new_collection
+    // -->
+
+    public static Map<String, MongoClient> connections = new HashMap<>();
+    public static Map<String, MongoDatabase> databases = new HashMap<>();
+    public static Map<String, MongoCollection<Document>> collections = new HashMap<>();
+
+    @Override
+    public void onDisable() {
+        for (Map.Entry<String, MongoClient> entry : connections.entrySet()) {
+            try {
+                entry.getValue().close();
+            }
+            catch (final Exception e) {
+                Debug.echoError(e);
+            }
+        }
+        connections.clear();
+        databases.clear();
+        collections.clear();
+    }
+
+    @Override
+    public void addCustomTabCompletions(TabCompletionsBuilder tab) {
+        tab.addWithPrefix("id:", connections.keySet());
+    }
+
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgPrefixed @ArgName("id") String id,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("connect") SecretTag uri,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("database") String database,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("collection") String collection,
+                                   // This is a boolean because it will let me check if it is present without using a prefix.
+                                   @ArgName("disconnect") boolean disconnect,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("command") MapTag command,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("find") MapTag findQuery,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("by_id") String findByID,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("insert") MapTag insert,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("update") MapTag oldData,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("new") MapTag newData,
+                                   @ArgName("upsert") boolean upsert,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("use_database") String newDatabase,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("use_collection") String newCollection) {
+        if (!CoreConfiguration.allowMongo) {
+            Debug.echoError(scriptEntry, "Mongo disabled by config!");
+            return;
+        }
+        if (id == null) {
+            Debug.echoError(scriptEntry, "Missing connection ID!");
+            return;
+        }
+        Runnable runnable = () -> {
+            try {
+                if (uri != null) {
+                    if (connections.containsKey(id.toLowerCase())) {

```

Comment:
`id` is `toLowerCase`ed every time it's used, should probably lowercase it once at the start (and should use `CoreUtilites#toLowerCase`)